File: runme/mcgen/local/datan/exasrc/e5mtx.f
      PROGRAM E5MTX
C demonstrates solution of matrix equation by Gauss algorithm
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      PARAMETER (N=3,M=3)
      DIMENSION A(N,N),B(N,M)
      DATA A/1.D0,2.D0,1.D0,2.D0,1.D0,1.D0,3.D0,-2.D0,2.D0/
      DATA B/1.D0,0.D0,0.D0,0.D0,1.D0,0.D0,0.D0,0.D0,1.D0/
C identify program to user
      WRITE(*,*)' Program E5MTX demonstrates use of MTXEQU'
      WRITE(*,*)' '
      WRITE(*,*)' Output produced by program E5MTX'
      WRITE(*,'(A)')' -----------------------------------------'
C write input data
      WRITE(*,'(A,I2,A,I2)')' N = ',N,', M = ',M
C write initial matrices
      WRITE(*,*)' A ='
      CALL MTXWRT(A,N,N)
      WRITE(*,*)' B ='
      CALL MTXWRT(B,N,M)
      WRITE(*,'(A)')' -----------------------------------------'
      WRITE(*,*)' '
C demonstrate MTXEQU
      CALL MTXEQU(A,B,N,M)
      WRITE(*,'(A)')' CALL MTXEQU(A,B,N,M) yields '
      WRITE(*,*)' B ='
      CALL MTXWRT(B,N,N)
      END

File: runme/mcgen/local/datan/exasrc/e5mtx.c
#include <stdio.h>
#include "datsrc.h"

/* Table of constant values */

static integer c__3 = 3;

main(void)
{
    /* Initialized data */

    static double a[9] /* was [3][3] */ = { 1.,2.,1.,2.,1.,1.,3.,-2., 2. },
                  b[9] /* was [3][3] */ = { 1.,0.,0.,0.,1.,0.,0.,0., 1. };

/* demonstrates solution of matrix equation by Gauss algorithm */
/* identify program to user */
    printf("%s\n\n", "Program E5MTX demonstrates use of MTXEQU.");
    printf("%s\n", "Output produced by program E5MTX.");
    printf("%s\n", "-----------------------------------------");
/* write input data */
    printf("%s%2li%s%2li\n", "N = ", c__3, ", M = ", c__3);
/* write initial matrices */
    printf("%s\n", "A =");
    mtxwrt_(a, &c__3, &c__3);
    printf("%s\n", "B =");
    mtxwrt_(b, &c__3, &c__3);
    printf("%s\n\n", "-----------------------------------------");
/* demonstrate MTXEQU */
    mtxequ_(a, b, &c__3, &c__3);
    printf("%s\n", "CALL MTXEQU(A,B,N,M) yields ");
    printf("%s\n", "B =");
    mtxwrt_(b, &c__3, &c__3);
    return 0;
} /* main */

